web_m2x_options/static/src/js/form.js

```diff
@@ -259,6 +314,54 @@ odoo.define('web_m2x_options.web_m2x_options', function (require) {
             });
 
             return def;
+        },
+
+        update_mru_list: function(){
+            var self = this,
+                mru_option = 'web_m2x_options_mru';
+            var key = self.compute_mru_key();
+
+            // check if the localstorage has some items for the current model
+            if (localStorage.getItem(mru_option)) {
+                var restore_mru_list = JSON.parse(localStorage.getItem(mru_option));
+                if (restore_mru_list[key]) {
+                    var queue = restore_mru_list[key];
+                    if (queue.indexOf(self.get_value(true)) < 0 && self.get_value(true)){
+                        if (queue.length < 5) {
+                            queue.push(self.get_value(true));
+                        }else {
+                            queue.shift();
+                            queue.push(self.get_value(true));
+                        }
+                        restore_mru_list[key] = queue;
+                    }
```

Comment:
> he most recurrent one should be on top 

Yes, otherwise your most frequently used one will drop off the list, ex with a size of 3

- A -> [ A ]
- B -> [ B, A ]
- A -> [ B, A ]
- C -> [ C, B, A ]
- A -> [ C, B, A ]
- D -> [ D, C, B ] and you've lost A even though you use it often